pkg/fleet/installer/packages/app_armor.go

```diff
@@ -20,30 +22,101 @@ import (
 )
 
 const (
-	appArmorConfigPath = "/etc/apparmor.d/abstractions/base.d"
-	appArmorProfile    = `/opt/datadog-packages/** rix,
-/proc/@{pid}/** rix,`
+	appArmorAbstractionDir       = "/etc/apparmor.d/abstractions/"
+	appArmorBaseProfile          = "/etc/apparmor.d/abstractions/base"
+	appArmorDatadogDir           = appArmorAbstractionDir + "datadog.d/"
+	appArmorInjectorProfilePath  = appArmorDatadogDir + "injector"
+	appArmorBaseDIncludeIfExists = "include if exists <abstractions/datadog.d>"
+	appArmorProfile              = `/opt/datadog-packages/** rix,
+/proc/@{pid}/** rix,
+/run/datadog/apm.socket rw,`
 )
 
-var datadogProfilePath = filepath.Join(appArmorConfigPath, "datadog")
+var datadogProfilePath = filepath.Join(appArmorDatadogDir, "datadog")
+
+func findAndReplaceAllInFile(filename string, needle string, replaceWith string) error {
+	haystack, err := os.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	haystack = []byte(strings.ReplaceAll(string(haystack), needle, replaceWith))
+
+	if err = os.WriteFile(filename, haystack, 0); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func unpatchBaseProfileWithDatadogInclude(filename string) error {
+
+	// make sure base profile exists before we continue
+	if _, err := os.Stat(filename); errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+
+	return findAndReplaceAllInFile(filename, "\n"+appArmorBaseDIncludeIfExists, "")
+}
+
+func patchBaseProfileWithDatadogInclude(filename string) error {
+	file, err := os.OpenFile(filename, os.O_APPEND|os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	// we'll be going through the whole base profile looking for a sign indicating this version
+	// supports the base.d extra profiles if it exists we'll return true
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), appArmorBaseDIncludeIfExists) {
+			return nil
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	// add a new line
+	if _, err = file.WriteString("\n" + appArmorBaseDIncludeIfExists); err != nil {
+		return err
+	}
```

Comment:
you should commit to disk with [file.Sync()](https://cs.opensource.google/go/go/+/refs/tags/go1.23.6:src/os/file_posix.go;l=162)